src/librustc_passes/ast_validation.rs

```diff
@@ -318,6 +310,13 @@ impl<'a> Visitor<'a> for AstValidator<'a> {
 
     fn visit_generics(&mut self, g: &'a Generics) {
         let mut seen_default = None;
+        for lt_def in &g.lifetimes {
```

Comment:
Lifetime definitions can happen in `for<'a, 'b, 'c>` in fn pointers, trait objects, etc.
It's better to place the error in `fn visit_lifetime_def`.